src/main/java/org/opentripplanner/middleware/persistence/TypedPersistence.java

```diff
@@ -150,6 +155,34 @@ public T getById(String id) {
         return mongoCollection.find(filter).limit(maximum).into(new ArrayList<>());
     }
 
+    /**
+     * Build a filter for querying Mongo based on userId and from/to dates.
+     */
+    public static Bson buildFilter(String userId, Date fromDate, Date toDate) {
```

Comment:
Please rename to something more descriptive like `filterByUserAndDate`.

Reply:
Updated.